[SecuritySolutions] Update risk score docs and add links to existing workflows

machadoum opened the pull request:
issue: https://github.com/elastic/kibana/issues/166717

## Summary
* Add the "How is risk score calculated?" button to the entity analytics page and risk score tab on the Host/User page.
* Add risk score hover action to the user/host overview component. We can't open a flyout here because this component is already inside a flyout. The links send the user to the complete documentation.

TODO
- [x] Update text with final version 
<img width="452" alt="Screenshot 2023-09-27 at 17 06 40" src="https://github.com/elastic/kibana/assets/1490444/d3237827-5982-4b4c-b01a-a5b2a6114176">


### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

machadoum committed:
Fix many risk score UI inconsistencies

elasticmachine commented:
Pinging @elastic/security-threat-hunting (Team:Threat Hunting)

machadoum commented:
@elasticmachine merge upstream

christineweng commented on x-pack/plugins/security_solution/public/flyout/right/components/host_entity_overview.tsx:
```diff
@@ -175,7 +175,7 @@ export const HostEntityOverview: React.FC<HostEntityOverviewProps> = ({ hostName
           <>
             {hostRiskData ? (
-              <RiskScore severity={hostRiskData.host.risk.calculated_level} />
+              <RiskScoreLevel severity={hostRiskData.host.risk.calculated_level} />
```
could you add the "learn more about risk score" link here as well?

SourinPaul replied:
Thanks for spotting the inconsistency.

christineweng commented on x-pack/plugins/security_solution/public/flyout/right/components/user_entity_overview.tsx:
```diff
@@ -174,7 +174,7 @@ export const UserEntityOverview: React.FC<UserEntityOverviewProps> = ({ userName
           <>
             {userRiskData ? (
-              <RiskScore severity={userRiskData.user.risk.calculated_level} />
+              <RiskScoreLevel severity={userRiskData.user.risk.calculated_level} />
```
same here could you add the "learn more about risk score" link?

christineweng commented on x-pack/plugins/security_solution/public/detections/pages/alert_details/tabs/summary/host_panel/index.tsx:
```diff
@@ -99,7 +99,7 @@ export const HostPanel = React.memo(
         : getEmptyTagValue();
       const hostRiskSeverity = hostRiskData ? (
-        <RiskScore severity={hostRiskData.host.risk.calculated_level} hideBackgroundColor />
+        <RiskScoreLevel severity={hostRiskData.host.risk.calculated_level} hideBackgroundColor />
```
should this have the "learn more" tooltip?

christineweng commented on x-pack/plugins/security_solution/public/detections/pages/alert_details/tabs/summary/user_panel/index.tsx:
```diff
@@ -78,7 +78,7 @@ export const UserPanel = React.memo(
         : getEmptyTagValue();
       const userRiskSeverity = userRiskData ? (
-        <RiskScore severity={userRiskData.user.risk.calculated_level} hideBackgroundColor />
+        <RiskScoreLevel severity={userRiskData.user.risk.calculated_level} hideBackgroundColor />
```
same here, "learn more" link is not visible

christineweng reviewed (commented):
@machadoum thanks for making these changes! 

The risk scores on alerts flyout (old and new) do not have the "learn more about risk score" tooltip. it will be best to stay consistent if this is the new standard

angorayc commented:
Hey @machadoum , the implementation looks fine, just found some UI inconsistency, LGTM as long as we confirmed them.
 
<img width="2547" alt="Screenshot 2023-09-23 at 21 02 10" src="https://github.com/elastic/kibana/assets/6295984/b18bf743-c855-47cb-a0ca-d346b8097e82">

In host overview we have no information icon:
<img width="2539" alt="Screenshot 2023-09-23 at 21 13 09" src="https://github.com/elastic/kibana/assets/6295984/bc925daa-296c-4806-95dc-1f15e6930d23">



The contents in the tooltips are different:

<img width="2273" alt="Screenshot 2023-09-23 at 21 18 49" src="https://github.com/elastic/kibana/assets/6295984/0054ee42-a304-46e9-bbbe-86e4417cbd99">



<img width="2547" alt="Screenshot 2023-09-23 at 21 19 27" src="https://github.com/elastic/kibana/assets/6295984/0b1dc9ec-7da3-41bd-a006-99546557d758">



In host risk table we have colour and risk level with a hover action `view unknown risk host`:

<img width="2536" alt="Screenshot 2023-09-23 at 21 30 15" src="https://github.com/elastic/kibana/assets/6295984/0475d3d1-147b-4071-a61b-ba01b972f66c">


In alerts table, the behaviour host risk classification seems to have different behaviour:
We could consider (in the future) display the column name as `Host risk classification` and have the same content rendered in the cell with an extra hover action.
<img width="2540" alt="Screenshot 2023-09-23 at 21 26 27" src="https://github.com/elastic/kibana/assets/6295984/dc090751-5319-4074-9c23-518017f57499">

machadoum commented:
> Hey @machadoum , the implementation looks fine, LGTM. Just found some UI inconsistency we could confirm them with PM and UX.
> 
> <img alt="Screenshot 2023-09-23 at 21 02 10" width="2000" src="https://user-images.githubusercontent.com/6295984/270131070-b18bf743-c855-47cb-a0ca-d346b8097e82.png">
> In host overview we have no information icon: <img alt="Screenshot 2023-09-23 at 21 13 09" width="2000" src="https://user-images.githubusercontent.com/6295984/270131360-bc925daa-296c-4806-95dc-1f15e6930d23.png">
> 
> The contents in the tooltips are different:
> 
> <img alt="Screenshot 2023-09-23 at 21 18 49" width="2000" src="https://user-images.githubusercontent.com/6295984/270131596-0054ee42-a304-46e9-bbbe-86e4417cbd99.png"> <img alt="Screenshot 2023-09-23 at 21 19 27" width="2000" src="https://user-images.githubusercontent.com/6295984/270131598-0b1dc9ec-7da3-41bd-a006-99546557d758.png">
> In host risk table we have colour and risk level with a hover action `view unknown risk host`:
> 
> <img alt="Screenshot 2023-09-23 at 21 30 15" width="2000" src="https://user-images.githubusercontent.com/6295984/270132074-0475d3d1-147b-4071-a61b-ba01b972f66c.png">
> In alerts table, the behaviour host risk classification seems to have different behaviour: We could consider (in the future) display the column name as `Host risk classification` and have the same content rendered in the cell with an extra hover action. <img alt="Screenshot 2023-09-23 at 21 26 27" width="2000" src="https://user-images.githubusercontent.com/6295984/270131984-dc090751-5319-4074-9c23-518017f57499.png">
> 
> We have host classification here but no hover action to filter by severity: <img alt="Screenshot 2023-09-23 at 21 48 11" width="2000" src="https://user-images.githubusercontent.com/6295984/270133030-a4824643-7253-426b-b181-64ae58d5bfef.png">

Hey Angela! Thank you for the attentive review. I created an [issue](https://github.com/elastic/kibana/issues/167143) for the reported UI inconsistencies and I will address them with Product and UX.

machadoum commented:
@elasticmachine merge upstream

christineweng commented:
> Hey @machadoum , the implementation looks fine, LGTM. Just found some UI inconsistency we could confirm them with PM and UX.

This is great catch @angorayc, the tech preview tooltip exists in the old alerts flyout (to see old flyout, add an alert to timeline and expand alert details in timeline - see screenshot below). At the time, risk score showed tech preview badge in some places, and we weren't sure if the badge can be dropped. To maintain consistency, a similar tech preview was added in the new expandable alerts view.

@machadoum appreciate you creating the ticket, let me know if you have any updates from product and UX :)

![image](https://github.com/elastic/kibana/assets/18648970/4a6205de-4ab7-4d35-b090-1f21c0330567)

SourinPaul commented:
@machadoum an additional item to track if we create a follow-up GH issue along with these inconsistencies:

* Changing the UI field label for `Host/ User Risk Classification` to `Host/User Risk Label`

We use the term `classification` in additional functionalities like 'asset classification'. Hence we decided to change the user-facing label to `label` to avoid confusion. cc: @craig-abbott 

Please lmk if you need additional contexts. Thanks!

machadoum commented:
@SourinPaul @christineweng Here is the ticket for inconsistencies: https://github.com/elastic/kibana/issues/167143
I will update it with the new findings.

machadoum commented:
@elasticmachine merge upstream

machadoum commented:
@christineweng I updated all places where risk score fields are displayed to have the tooltip (except for tables). Could you please take a second look? I also extracted the tooltip content to a component so we can reuse it.

<img width="431" alt="Screenshot 2023-09-27 at 11 53 06" src="https://github.com/elastic/kibana/assets/1490444/31a8401c-6bd6-4d79-8c0b-cda69ae9044a">
<img width="428" alt="Screenshot 2023-09-27 at 11 53 14" src="https://github.com/elastic/kibana/assets/1490444/afad0461-6fd6-47f6-8755-b8b7fda6df42">
<img width="1328" alt="Screenshot 2023-09-27 at 11 53 38" src="https://github.com/elastic/kibana/assets/1490444/6b9d4224-3284-4bf6-8903-8c95ddb4ea00">
<img width="1318" alt="Screenshot 2023-09-27 at 11 54 00" src="https://github.com/elastic/kibana/assets/1490444/bd81fc6f-d8f8-4f02-a141-9ff47528b375">
<img width="595" alt="Screenshot 2023-09-27 at 13 48 55" src="https://github.com/elastic/kibana/assets/1490444/a7640082-4d1a-4642-8121-c1ca50f48805">

christineweng reviewed (approved):
LGTM 🚀 Thank you for making the changes!

nkhristinin reviewed (approved):
Detection-engine changes look good to me, but I am confused in the text that we query "open" alerts, and I am not sure that it's true for new algo.

@rylnd do we have this filter for alerts?

rylnd commented:
> @rylnd do we have this filter for alerts?

We do now: https://github.com/elastic/kibana/pull/167460/commits/be1d8440e7fff4b609c4cf5d7e97592e72f8ac87

machadoum commented:
The changes in this PR have widely diverged from the code reviewed code. I am closing this PR and opening a new one that on;y contains the new changes.
